vault/expiration.go

```diff
@@ -1812,6 +1820,7 @@ func (m *ExpirationManager) updatePendingInternal(le *leaseEntry) {
 
 		m.removeFromPending(m.quitContext, le.LeaseID, false)
 		m.irrevocable.Store(le.LeaseID, m.inMemoryLeaseInfo(le))
+		m.irrevocableLeaseCount++
```

Comment:
we might want to qualify this based on `!leaseInIrrevocable` to avoid potentially over counting.

On that note, we could be slightly more efficient/clear by also qualifying `m.removeFromPending` on `leaseInPending`.

I'm not sure if we should qualify `m.irrevocable.Store()` on `!leaseInIrrevocable` as it might be possible this call is being used to update in-memory data (something to look at, or ignore for now).

These last two points are less important, but it looks like we want to qualify the increment count on if it's being added to the `irrevocable` map

Reply:
Roger that, qualifying the increment makes sense! I'm thinking those other 2 points can be ignored for now, as they would both be no-op functions if the data isn't present in pending / is present in irrevocable maps, so they should be safe without the qualifying checks.

Reply:
2558e12